src/librustc/traits/object_safety.rs

```diff
@@ -189,10 +190,14 @@ fn object_safety_violations_for_trait(
                         tcx.def_path_str(trait_def_id)
                     ),
                 );
-                match violation.span() {
-                    Some(span) => err.span_label(span, violation.error_msg()),
-                    None => err.note(&violation.error_msg()),
-                };
+                let spans = violation.spans();
```

Comment:
This code snippet seems to be a recurring thing, does it warrant its own method on `err`?

Reply:
This comment made me realize that one of the two isn't needed.